File: android/app/src/main/java/my/ssdid/wallet/domain/revocation/StatusListCredential.kt
package my.ssdid.wallet.domain.revocation

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import my.ssdid.wallet.domain.model.Proof

@Serializable
data class StatusListCredential(
    @SerialName("@context") val context: List<String> = emptyList(),
    val id: String? = null,
    val type: List<String>,
    val issuer: String,
    val credentialSubject: StatusListSubject,
    val proof: Proof? = null
)

@Serializable
data class StatusListSubject(
    val type: String,
    val statusPurpose: String,
    val encodedList: String
)

File: sdk/ios/SsdidCore/Sources/SsdidCore/Domain/Revocation/StatusListCredential.swift
import Foundation

public struct StatusListCredential: Codable {
    public let context: [String]
    public let id: String?
    public let type: [String]
    public let issuer: String
    public let credentialSubject: StatusListSubject
    public let proof: Proof?

    enum CodingKeys: String, CodingKey {
        case context = "@context"
        case id, type, issuer, credentialSubject, proof
    }

    public init(
        context: [String] = [],
        id: String? = nil,
        type: [String],
        issuer: String,
        credentialSubject: StatusListSubject,
        proof: Proof? = nil
    ) {
        self.context = context
        self.id = id
        self.type = type
        self.issuer = issuer
        self.credentialSubject = credentialSubject
        self.proof = proof
    }
}

public struct StatusListSubject: Codable {
    public let type: String
    public let statusPurpose: String
    public let encodedList: String
}
